server/channels/app/platform/metrics.go

```diff
@@ -166,9 +178,44 @@ func (pm *platformMetrics) initMetricsRouter() error {
 	pm.router.Handle("/debug/pprof/threadcreate", pprof.Handler("threadcreate"))
 	pm.router.Handle("/debug/pprof/block", pprof.Handler("block"))
 
+	// Plugins metrics route
+	pluginsMetricsRoute := pm.router.PathPrefix("/plugins/{plugin_id:[A-Za-z0-9\\_\\-\\.]+}/metrics").Subrouter()
+	pluginsMetricsRoute.HandleFunc("", pm.servePluginMetricsRequest)
+	pluginsMetricsRoute.HandleFunc("/{anything:.*}", pm.servePluginMetricsRequest)
+
 	return nil
 }
 
+func (pm *platformMetrics) servePluginMetricsRequest(w http.ResponseWriter, r *http.Request) {
+	pluginID := mux.Vars(r)["plugin_id"]
+
+	pluginsEnvironment := pm.getPluginsEnv()
+	if pluginsEnvironment == nil {
+		err := model.NewAppError("ServePluginMetricsRequest", "app.plugin.disabled.app_error",
+			nil, "Enable plugins to serve plugin metric requests", http.StatusNotImplemented)
+		mlog.Error(err.Error())
+		w.WriteHeader(err.StatusCode)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(err.ToJSON()))
+		return
+	}
+
+	hooks, err := pluginsEnvironment.HooksForPlugin(pluginID)
+	if err != nil {
+		mlog.Debug("Access to route for non-existent plugin",
+			mlog.String("missing_plugin_id", pluginID),
+			mlog.String("url", r.URL.String()),
+			mlog.Err(err))
+		http.NotFound(w, r)
+		return
+	}
+
+	subpath, _ := utils.GetSubpathFromConfig(pm.cfgFn())
+	r.URL.Path = strings.TrimPrefix(r.URL.Path, path.Join(subpath, "plugins", pluginID, "metrics"))
+
+	hooks.ServeMetrics(&plugin.Context{}, w, r)
```

Comment:
Passing an empty context for the time being, to be decided whether we should support forms of authentication. My guess is no but we could still pass the session information if we can see a use case.

Reply:
0/5 maybe it's worth adding these thoughts as a comment

Reply:
That's a good idea, thanks :+1: